### DISCUSSION TITLE
Set Custom Timeout Value

### DISCUSSION THREAD

**Discussion by khaliduzzaman-mredul** [davesteele/comitup#226] [Q&A]
If the raspberry pi doesn't find any known network it enters access point mode. And after 3 minutes it searches for network again.
My question is, how can I set that 3 minutes value to something else?
Any suggestions would be very helpful
Thank you!

**Comment by davesteele** [comment]
You can, but it involves editing the [source](https://github.com/davesteele/comitup/blob/1cef19ff2e50cea1f16061022ec162136167115d/comitup/states.py#L29). 

A couple of notes:

- It's not always 3 minutes. If a previous connection attempt fails, it is currently [6 minutes](https://github.com/davesteele/comitup/blob/1cef19ff2e50cea1f16061022ec162136167115d/comitup/states.py#L230), for [reasons](https://github.com/davesteele/comitup/commit/f96553030c9a7607c0e766beb4d91c2e95a62b5f).
- The timeout value serves other functions, such as terminating an attempt to establish HOTSPOT mode, and defining a check time for verifying that we are still CONNECTED. See the *_timeout() functions in states.py

**Reply by khaliduzzaman-mredul** [reply]
Hello mr steele. Thanks a lot for the reply. 
What if I install comitup using pip?
Where do I get the states.py in my raspberry pi?

**Reply by davesteele** [reply]
That depends on how you are invoking pip. It is stored with the comitup module.

The deb package location is /usr/share/comitup/comitup/states.py

Be aware that a pip install requires [additional work](https://github.com/davesteele/comitup/wiki/Installing-Comitup), and that there are OS package dependencies that a pip install won't consider.

**Reply by khaliduzzaman-mredul** [reply]
Hello Mr. Steele. 
Is there any known issues or update about comitup for the new version of raspbian? It seems like not working after the latest raspbian update.

**Reply by davesteele** [reply]
[Yes](https://github.com/davesteele/comitup/issues/234). The latest comitup image works for now.
